src/axom/inlet/Field.cpp

```diff
@@ -58,12 +60,168 @@ bool Field::required()
+    setWarningFlag(m_sidreRootGroup);
+  }
+  if (m_sidreGroup->hasView("defaultValue")) {
+    std::string msg = fmt::format("Inlet Field has already defined default value: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewScalar("defaultValue", value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewScalar("value", value);
+    }
+  }
+  
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(double value) {
+  if (m_type != axom::sidre::DataTypeId::DOUBLE_ID) {
+    SLIC_WARNING("Field value type did not match DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  if (m_sidreGroup->hasView("defaultValue")) {
+    std::string msg = fmt::format("Inlet Field has already defined default value: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewScalar("defaultValue", value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewScalar("value", value);
+    }
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::range(double startVal, double endVal) {
+  if (m_type != axom::sidre::DataTypeId::DOUBLE_ID) {
+    SLIC_WARNING("Field value type did not match DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+  }
+
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")) {
+    std::string msg = fmt::format("Inlet Field has already defined range: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    double tuple[2];
+    tuple[0] = startVal;
+    tuple[1] = endVal;
+    auto view = m_sidreGroup->createViewAndAllocate("range", axom::sidre::DOUBLE_ID, 2);
+    view->getBuffer()->copyBytesIntoBuffer(tuple, 2*sizeof(double));
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::range(int startVal, int endVal) {
+  if (m_type != axom::sidre::DataTypeId::INT_ID) {
+    SLIC_WARNING("Field value type did not match INT");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")) {
+    std::string msg = fmt::format("Inlet Field has already defined range: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    int tuple[2];
+    tuple[0] = startVal;
+    tuple[1] = endVal;
+    auto view = m_sidreGroup->createViewAndAllocate("range", axom::sidre::INT_ID, 2);
+    view->getBuffer()->copyBytesIntoBuffer(tuple, 2*sizeof(int));
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::validValues(std::vector<int> set) {
```

Comment:
I didn't get there yet, but is there a check that the default value is within the specified range?

Reply:
Yep, Inlet::verify() takes care of that

Reply:
Not sure that is true.  It will check if the value is not in the input deck and there is a default value (because it will set value to the default value at that point i think) but if you have a value in the input deck AND specify a default value, it doesn't specifically check the default value.

The case that needs to be checked is this (and add a test for it please):

// input deck = "foo=5"

inlet()->addInt("foo")->defaultValue(6)->validValues({5}); // should throw a warning

Reply:
Sorry, ^ that's right I misread the intial question. I'll add a check specifically for the default value.